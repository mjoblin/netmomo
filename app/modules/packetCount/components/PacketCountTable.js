import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'antd';


const PacketCountTable = ({ packetCounts }) => {
    const columns = [
        {
            title: 'Protocol',
            dataIndex: 'protocol',
            key: 'protocol',
            sorter: (a, b) => a.protocol.localeCompare(b.protocol),
            width: 100,
        },
        {
            title: 'Count',
            dataIndex: 'count',
            key: 'count',
            sorter: (a, b) => a.count - b.count,
            render: text => text.toLocaleString(),
            className: 'column-count',
            width: 100,
        },
    ];

    return (
        <Table
            dataSource={packetCounts}
            columns={columns}
            pagination={false}
            size="middle"
        />
    );
};


PacketCountTable.propTypes = {
    packetCounts: PropTypes.object.isRequired,
};


export default PacketCountTable;
