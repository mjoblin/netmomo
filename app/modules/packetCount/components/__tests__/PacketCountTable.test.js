import React from 'react';
import { shallow } from 'enzyme';
import { PacketCountTable } from '../PacketCountTable';
import testData from './testData.json';

describe('PacketCountTable', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <PacketCountTable
                packetCounts={testData}
            />
        );

        expect(component).toMatchSnapshot();
    });
});