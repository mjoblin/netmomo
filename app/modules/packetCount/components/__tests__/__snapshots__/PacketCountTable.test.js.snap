// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PacketCountTable matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PacketCountTable
    packetCounts={
      Array [
        Object {
          "count": 94410,
          "key": "Ethernet",
          "protocol": "Ethernet",
        },
        Object {
          "count": 93325,
          "key": "IP",
          "protocol": "IP",
        },
        Object {
          "count": 94243,
          "key": "UDP",
          "protocol": "UDP",
        },
        Object {
          "count": 94289,
          "key": "Raw",
          "protocol": "Raw",
        },
        Object {
          "count": 987,
          "key": "IPv6",
          "protocol": "IPv6",
        },
        Object {
          "count": 98,
          "key": "ARP",
          "protocol": "ARP",
        },
        Object {
          "count": 17,
          "key": "Padding",
          "protocol": "Padding",
        },
        Object {
          "count": 69,
          "key": "TCP",
          "protocol": "TCP",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "bordered": false,
      "className": "",
      "columns": Array [
        Object {
          "dataIndex": "protocol",
          "key": "protocol",
          "sorter": [Function],
          "title": "Protocol",
        },
        Object {
          "className": "column-count",
          "dataIndex": "count",
          "key": "count",
          "render": [Function],
          "sorter": [Function],
          "title": "Count",
          "width": 100,
        },
      ],
      "dataSource": Array [
        Object {
          "count": 94410,
          "key": "Ethernet",
          "protocol": "Ethernet",
        },
        Object {
          "count": 93325,
          "key": "IP",
          "protocol": "IP",
        },
        Object {
          "count": 94243,
          "key": "UDP",
          "protocol": "UDP",
        },
        Object {
          "count": 94289,
          "key": "Raw",
          "protocol": "Raw",
        },
        Object {
          "count": 987,
          "key": "IPv6",
          "protocol": "IPv6",
        },
        Object {
          "count": 98,
          "key": "ARP",
          "protocol": "ARP",
        },
        Object {
          "count": 17,
          "key": "Padding",
          "protocol": "Padding",
        },
        Object {
          "count": 69,
          "key": "TCP",
          "protocol": "TCP",
        },
      ],
      "indentSize": 20,
      "loading": false,
      "locale": Object {},
      "pagination": false,
      "prefixCls": "ant-table",
      "rowKey": "key",
      "rowSelection": null,
      "showHeader": true,
      "size": "middle",
      "useFixedHeader": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "bordered": false,
        "className": "",
        "columns": Array [
          Object {
            "dataIndex": "protocol",
            "key": "protocol",
            "sorter": [Function],
            "title": "Protocol",
          },
          Object {
            "className": "column-count",
            "dataIndex": "count",
            "key": "count",
            "render": [Function],
            "sorter": [Function],
            "title": "Count",
            "width": 100,
          },
        ],
        "dataSource": Array [
          Object {
            "count": 94410,
            "key": "Ethernet",
            "protocol": "Ethernet",
          },
          Object {
            "count": 93325,
            "key": "IP",
            "protocol": "IP",
          },
          Object {
            "count": 94243,
            "key": "UDP",
            "protocol": "UDP",
          },
          Object {
            "count": 94289,
            "key": "Raw",
            "protocol": "Raw",
          },
          Object {
            "count": 987,
            "key": "IPv6",
            "protocol": "IPv6",
          },
          Object {
            "count": 98,
            "key": "ARP",
            "protocol": "ARP",
          },
          Object {
            "count": 17,
            "key": "Padding",
            "protocol": "Padding",
          },
          Object {
            "count": 69,
            "key": "TCP",
            "protocol": "TCP",
          },
        ],
        "indentSize": 20,
        "loading": false,
        "locale": Object {},
        "pagination": false,
        "prefixCls": "ant-table",
        "rowKey": "key",
        "rowSelection": null,
        "showHeader": true,
        "size": "middle",
        "useFixedHeader": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
